src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -211,6 +230,24 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
               },
             );
           }
+
+          // for each question and answer, see if we find a matching concept, and, if so:
+          //   1) replace the concept reference with uuid (for the case when the form references the concept by mapping)
+          //   2) use the concept display as the label if no label specified
+          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
+          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
+          field.label = field.label ? field.label : matchingConcept?.display;
+          if (field.questionOptions.answers) {
+            field.questionOptions.answers = field.questionOptions.answers.map(answer => {
+              const matchingAnswerConcept = findConceptByReference(answer.concept, concepts);
+              return {
+                ...answer,
+                concept: matchingAnswerConcept ? matchingAnswerConcept.uuid : answer.concept,
+                label: answer.label ? answer.label : matchingAnswerConcept?.display,
+              };
+            });
+          }
```

Comment:
Ah, thanks @samuelmale for pointing that fetch out, I hadn't noticed it.  I can see if I we can replace it with the new fetch.

Reply:
I'm terrible at naming things but I was thinking introducing a `meta` (for the lack of a better name) property at the question root level. This can be used to hold a map of arbitrary objects associated to a field.


I don't object using the existing `field.questionOptions.concept` property but I think we need a meta-like property to keep things in one place.

Reply:
So I liked the idea of meta a lot, but when I started to think about designing it out, I had doubts.  Are you thinking adding something like this:

```
export interface OHRIFormQuestionOptions {
  **meta?: object;**
  extensionId?: string;
  extensionSlotName?: string;
  rendering: RenderType;
  concept?: string;
  /**
   * max and min are used to validate number field values
   */
  max?: string;
  min?: string;
  /**
   * maxLength and maxLength are used to validate text field length
   */
  maxLength?: string;
  minLength?: string;
  showDate?: string;
  answers?: Array<Record<any, any>>;
  weeksList?: string;
  locationTag?: string;
  rows?: number;
  toggleOptions?: { labelTrue: string; labelFalse: string };
  repeatOptions?: { addText?: string; limit?: string; limitExpression?: string };
  defaultValue?: any;
  calculate?: {
    calculateExpression: string;
  };
  isDateTime?: { labelTrue: boolean; labelFalse: boolean };
  usePreviousValueDisabled?: boolean;
}
```

When I started to map this out, the I realized that aren't a lot of the existing fields stuff we'd consider "metadata" so it might get confusing?  Also, if we store all the concepts for the questions and answers within this "meta" section, we still need to figure out a way to map the answer concepts to the specific answers they go with?

Which led me to think... maybe worth starting a broader discussion about this? Is there a place where we are having Form Engine design discussions?  Either Talk, or we could look into the RPC model they were using for other O3 design discussion?

Reply:
>Also, if we store all the concepts for the questions and answers within this "meta" section, we still need to figure out a way to map the answer concepts to the specific answers they go with?

I didn't think of a use case for linking the concept answers; if you asked me, I would say it's less likely to be desired. That being said, mapping ceases to be an issue.

I see in the proposed typings above you included `meta` to the question options, I was inclined to have it at the root level instead:

```ts

export interface OHRIFormField {
  label: string;
  type: string;
  questionOptions: OHRIFormQuestionOptions;
  id: string;
  // other props...
  meta: Record<string, any>;
}
```

PS: I'm honestly not so dogmatic about how we should approach this. I will let others share their thoughts; I just don't want to drag this.. cc: @ibacher, @denniskigen etc..

Reply:
Thanks Samuel... yes, let's see what Ian and Dennis think... that being said, if it becomes a dragged out discussion, I don't think blocks this PR?  We could merge this in and handle adding "meta" as a future ticket?